source/effect_parser.cpp

```diff
@@ -873,7 +873,7 @@ bool reshadefx::parser::parse_expression_unary(expression &exp)
 
 				arguments[i].add_cast_operation(param_type);
 
-				if (symbol.op == symbol_type::function || param_type.has(type::q_out))
```

Comment:
This is necessary and for compatibility with the SPIR-V code generator.

Reply:
I am sorry I do not know what is the problem, so I simply reverted it.